src/Apps/Legion/Services/AppTypeUtilities.ts

```diff
@@ -0,0 +1,162 @@
+import {
+    IDbEntity,
+    IDbProperty,
+    IDbRelationship,
+    IDbRelationshipType,
+    IDbType,
+    IViewEntity,
+    IViewProperty,
+    IViewRelationship,
+    IViewRelationshipType,
+    IViewType
+} from '../Models';
+
+// #region Entities
+export function convertEntityToDb(viewModel: IViewEntity): IDbEntity {
+    return {
+        friendlyName: viewModel.friendlyName,
+        id: viewModel.id,
+        sourceConnectionString: viewModel.sourceConnectionString,
+        sourceEntityId: viewModel.sourceEntityId,
+        values: viewModel.values, // TODO: do we need to serialize each property value into a string?
+        typeId: viewModel.type.id
+    };
+}
+
+export function convertEntityToView(
+    dbModel: IDbEntity,
+    state: {
+        types: IDbType[];
+        properties: IDbProperty[];
+    }
+): IViewEntity {
+    const type = state.types.find((x) => x.id === dbModel.typeId);
+    return {
+        friendlyName: dbModel.friendlyName,
```

Comment:
same comment for the following methods to use spread method to copy the same fields to the new object except some field